SparkPost/2web2ui
# FE-368 Track form events

This WIP, tracks redux form events to Google Analytics via GTM. Currently, GTM and GA tst is configured to accept events. 

It tracks redux form is initialization, form validation error/success, form submission. Initially I've tracked with redux-beacon extension for GTM but seems that's unnecessary (replaced that with `pushEvents`). 

Todos:

- [x] Remove beacon logger
- [x] Track non-redux form
  - I guess we can just call `pushEvents` or make a wrapper like `trackEvent`.
- [x] Validate if these tracking events are measurable 

Any early feedbacks are welcome.

Update:
1) Somewhat based on Bryan's suggestion, I've setting validation `label` like `<formName> JSON.stringify(<validation_errors>)`. We can use `regex` to filter by form name and also then what error actually occurred. 

2) could not test `eventsMap` the way i wanted. I wanted to mock `getFormDefinition` and assert it has been called with correct arg (like 'Initialize' for first item). 


Deployment tasks:
**TST:** 
- [x] Setup GTM (setup variables, triggers and tags in an appropriate folder)
- [ ] Ship Code
- [ ] Live Preview 
- [ ] Publish GTM Changes

**STG:** 
- [ ] Setup GTM (setup variables, triggers and tags in an appropriate folder)
- [ ] Ship Code
- [ ] Live Preview 
- [ ] Publish GTM Changes

**PRD** 
- [ ] Setup GTM (setup variables, triggers and tags in an appropriate folder)
- [ ] Ship Code
- [ ] Live Preview 
- [ ] Publish GTM Changes


**Testing**
- Start Preview Mode in GTM
- Start using any (redux) form normally. The events should be sent to GA via GTM. Check GTM preview/data layer for event details
